pkg/crcontrollers/reservation/reservationcontroller.go

```diff
@@ -0,0 +1,136 @@
+package reservation
+
+import (
+	"context"
+	"time"
+
+	"github.com/sirupsen/logrus"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	v1api "github.com/dell/csi-baremetal/api/generated/v1"
+	v1 "github.com/dell/csi-baremetal/api/v1"
+	acrcrd "github.com/dell/csi-baremetal/api/v1/acreservationcrd"
+	"github.com/dell/csi-baremetal/pkg/base/capacityplanner"
+	"github.com/dell/csi-baremetal/pkg/base/k8s"
+	metrics "github.com/dell/csi-baremetal/pkg/metrics/common"
+)
+
+const (
+	contextTimeoutSeconds = 60
+)
+
+// Controller to reconcile aviliablecapacityreservation custom resource
+type Controller struct {
+	client                 *k8s.KubeClient
+	log                    *logrus.Entry
+	capacityManagerBuilder capacityplanner.CapacityManagerBuilder
+}
+
+// NewController creates new instance of Controller structure
+// Receives an instance of base.KubeClient, node ID and logrus logger
+// Returns an instance of Controller
+func NewController(client *k8s.KubeClient, log *logrus.Logger) *Controller {
+	return &Controller{
+		client:                 client,
+		log:                    log.WithField("component", "ReservationController"),
+		capacityManagerBuilder: &capacityplanner.DefaultCapacityManagerBuilder{},
+	}
+}
+
+// SetupWithManager registers Controller to ControllerManager
+func (c *Controller) SetupWithManager(mgr ctrl.Manager) error {
+	return ctrl.NewControllerManagedBy(mgr).
+		For(&acrcrd.AvailableCapacityReservation{}).
+		Complete(c)
+}
+
+// Reconcile reconciles ACR custom resources
+func (c *Controller) Reconcile(req ctrl.Request) (ctrl.Result, error) {
+	defer metrics.ReconcileDuration.EvaluateDurationForType("reservation_controller")()
+
+	ctx, cancelFn := context.WithTimeout(context.Background(), contextTimeoutSeconds*time.Second)
+	defer cancelFn()
+
+	// read name
+	name := req.Name
+	// customize logging
+	log := c.log.WithFields(logrus.Fields{"method": "Reconcile", "name": name})
+
+	// obtain corresponding reservation
+	reservation := &acrcrd.AvailableCapacityReservation{}
+	if err := c.client.ReadCR(ctx, name, "", reservation); err != nil {
+		log.Warningf("Failed to read available capacity reservation %s CR", name)
+		return ctrl.Result{}, client.IgnoreNotFound(err)
+	}
+
+	log.Debugf("Reservation changed: %v", reservation)
+	return c.handleReservationUpdate(ctx, log, reservation)
+}
+
+func (c *Controller) handleReservationUpdate(ctx context.Context, log *logrus.Entry,
```

Comment:
You can use logger from Controller without paremeter in function

Reply:
It's customized with the reservation name

Reply:
Do you suggest to customize again?

Reply:
Usually we customize it in every function. But your approach is ok